src/controller/python/chip/storage/__init__.py

```diff
@@ -0,0 +1,159 @@
+#
+#    Copyright (c) 2021 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+# Needed to use types in type hints before they are fully defined.
+from __future__ import annotations
+
+import ctypes
+from dataclasses import dataclass, field
+from typing import *
+from ctypes import *
+from rich.pretty import pprint
+import ipdb
+import json
+import logging
+import base64
+import chip.exceptions
+import copy
+import chip.native
+import builtins
+
+_SyncSetKeyValueCbFunct = CFUNCTYPE(
+    None, py_object, c_char_p, POINTER(c_char),  c_uint16)
+_SyncGetKeyValueCbFunct = CFUNCTYPE(
+    None, py_object, c_char_p, POINTER(c_char), POINTER(c_uint16))
+_SyncDeleteKeyValueCbFunct = CFUNCTYPE(None, py_object, POINTER(c_char_p))
+
+
+@_SyncSetKeyValueCbFunct
+def _OnSyncSetKeyValueCb(storageObj, key: str, value, size):
+    storageObj.SetSdkKey(key.decode("utf-8"), ctypes.string_at(value, size))
+
+
+@_SyncGetKeyValueCbFunct
+def _OnSyncGetKeyValueCb(storageObj, key: str, value, size):
+    try:
+        keyValue = storageObj.GetSdkKey(key.decode("utf-8"))
+    except Exception as ex:
+        keyValue = None
+
+    if (keyValue):
+        if (size[0] < len(keyValue)):
+            size[0] = len(keyValue)
+            return
+
+        count = 0
+
+        for idx, val in enumerate(keyValue):
+            value[idx] = val
+            count = count + 1
+
+        size[0] = count
+    else:
+        size[0] = 0
+
+
+@_SyncDeleteKeyValueCbFunct
+def _OnSyncDeleteKeyValueCb(storageObj, key):
+    print("OnSyncDeleteKeyValue")
+    storageObj.SetKey(key, None)
```

Comment:
PersistentStorage does not has `SetKey`, seems it should be `SetSdkKey`?
Also, the type of key is bytes instead of str